jsonnet/kube-prometheus/components/prometheus-adapter.libsonnet

```diff
@@ -27,7 +27,7 @@ local defaults = {
     resourceRules: {
       cpu: {
         containerQuery: 'sum(irate(container_cpu_usage_seconds_total{<<.LabelMatchers>>,container!="",pod!=""}[5m])) by (<<.GroupBy>>)',
-        nodeQuery: 'sum(1 - irate(node_cpu_seconds_total{mode="idle"}[5m]) * on(namespace, pod) group_left(node) node_namespace_pod:kube_pod_info:{<<.LabelMatchers>>}) by (<<.GroupBy>>)',
+        nodeQuery: 'sum(1 - irate(node_cpu_seconds_total{mode="idle"}[5m]) * on(namespace, pod) group_left(node) node_namespace_pod:kube_pod_info:{<<.LabelMatchers>>}) by (<<.GroupBy>>) or sum (1- irate(windows_cpu_time_total{mode="idle"}[5m])) by (<<.GroupBy>>)',
```

Comment:
Correct me if I am wrong, but I don't think the `windows_cpu_time_total` metric as a `node` label that we could use in this query to match the resource usage to a particular node. If that's the case, we might also need to correlate the `windows-exporter` instance to its node as it is done for `node-exporter` in the first part of the query.

Reply:
unfortunately windows_exporter does not run as a pod but rather as a Windows service on the node, how do we correlate it using `node_namespace_pod:kube_pod_info` as in case of node-exporter?

Reply:
If I understand correctly, it is not directly tied to any Kubernetes resource right? Is there any metrics/recording rules that would link the service to the node it is running on?

Reply:
I can add a screenshot here, I took for the windows_cpu_time_total query metric, and I can see it is tied to the node it is running on and that can work as a correlation, does that seem right?
![image](https://user-images.githubusercontent.com/35242579/112514843-17c98800-8d6c-11eb-9bb9-d47fdf61bee2.png)

Reply:
Yes that's perfect, I thought the label wasn't present as it wasn't mentioned in the exporter doc. Just to be sure, do you know if the `node` label is specific to OpenShift or if it also present in vanilla Kubernetes installations?

While you are at it, would you mind also checking if the `windows_memory_available_bytes` and `windows_cs_physical_memory_bytes` metrics have an `instance` label?

Reply:
yes I can verify windows_memory_available_bytes and windows_cs_physical_memory_bytes have the instance label too. I don't think the label is specific to OpenShift since we do not do much other than building the exporter from upstream as a part of our payload and running the built .exe on the Windows node, but let me confirm that once.

Reply:
ok so on a closer look I see that the windows_exporter returns node IP and not node names but we need node names in order to display node graphs on OpenShift console due to the way the console queries are formulated that is why replace the node IP with node name [here](https://github.com/openshift/windows-machine-config-operator/blob/f19f07279865d80f9e58d0f326d0dbd7e42003c5/deploy/olm-catalog/windows-machine-config-operator/manifests/service-monitor.yaml#L14), this makes it OpenShift specific.

Reply:
Thank you for doing all the research! :bow: 

From your screenshot, I would say that both the `instance` and `node` labels contain the node name. Even though they look like IP addresses, I think they are also the node names.

We can't update the CPU override as it is required by the current query. The current query doesn't have an `instance` label so we need to use the `node` one. As for the current memory query, it has an `instance` label but not a `node` one.

Reply:
That being said, the queries look good to me. Would you just mind adding `{job="windows-exporter",<<.LabelMatchers>>}` to the metrics?

Reply:
I wonder if we could also add the ServiceMonitor configuration shared by @mansikulkarni96 to the windows add-on as it will be required for the memory query to work. wydt @paulfantom @jsturtevant?

Reply:
Adding the configuration to service monitor sounds good to me. I believe that will set the node names as the label? That would be preferable especially for adding support for the memory query.

>  unfortunately windows_exporter does not run as a pod but rather as a Windows service on the node

One other thing to mention is that we have been working on adding [HostProcess container](https://github.com/kubernetes/enhancements/issues/1981) support (similar to Linux Privileged containers) to Windows which would enable us to run windows_exporter as a pod.  We plan to have alpha support in 1.22 which is several months away but is there any considerations for this query from a backwards compatibility once we get there?

Reply:
@dgrisonnet I have updated the queries with the label and also added the serviceMonitor to the add-on's. Could you let me know if that is the right place to add it or I should move it somewhere else, thanks!
@jsturtevant I haven't added any service or pod specific fields to the queries, they only use the windows_exporter metrics names and job fields currently.

Reply:
Actually, taking into account the current configuration, it might be better to just add the relabeling config to the scrape config of windows-exporter instead of introducing a ServiceMonitor: https://github.com/prometheus-operator/kube-prometheus/blob/main/jsonnet/kube-prometheus/addons/windows.libsonnet#L6-L13

```jsonnet
relabel_configs: [
  {
    action: "replace",
    regex: "(.*)",
    replacement: "$1",
    sourceLabels: [
      "__meta_kubernetes_endpoint_address_target_name",
    ],
    targetLabel: "instance",
  },
],
```

> Adding the configuration to service monitor sounds good to me. I believe that will set the node names as the label? That would be preferable especially for adding support for the memory query.

It will set the node names in the `instance` label that is required by the query.

Reply:
Thanks for taking a look @dgrisonnet , I have updated the PR with you suggestion please take a look.